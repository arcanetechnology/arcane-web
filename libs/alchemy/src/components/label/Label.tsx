/** @format */

import { FlowComponent, JSX, splitProps, mergeProps } from 'solid-js';

type LabelProps = JSX.LabelHTMLAttributes<HTMLLabelElement>;

export const Label: FlowComponent<LabelProps> = (props) => {
  const merged = mergeProps({ id: 'arcane' }, props);
  const [local, others] = splitProps(merged, ['children', 'id']);
  return (
    <label {...others} id={`${local.id}-label`}>
      {local.children}
    </label>
  );
};
