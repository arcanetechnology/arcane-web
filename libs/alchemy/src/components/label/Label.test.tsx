/** @format */

import { Label } from './Label';
import { cleanup, render } from 'solid-testing-library';

describe('alchemy label', () => {
  afterEach(cleanup);

  it('renders label element', () => {
    const tree = render(() => <Label>test-label</Label>).baseElement;
    expect(tree).toMatchSnapshot();
  });
});
