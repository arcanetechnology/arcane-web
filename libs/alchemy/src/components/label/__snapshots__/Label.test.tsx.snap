// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alchemy label renders label element 1`] = `
<body>
  <div>
    <label
      id="arcane-label"
    >
      test-label
    </label>
  </div>
</body>
`;
